### PR TITLE
SDP-1922: Consolidate wallet data retrieval under `/profile` endpoint

### PR DESCRIPTION
**Pull request by JiahuiWho**
### What

Consolidated the `embedded_wallet_profile_handler` so it fetches verification state, supported assets, and receiver contact, with a shared helper for translating service errors into 401/500 responses.

### Why

we can gate everything off the `isPending` flag,  and only fetch wallet details when verification already passed. This keeps the response aligned with the flow the client has to follow, and avoids partial states.

### Known limitations

N/A

### Checklist

- [ ] Title follows `SDP-1234: Add new feature` or `Chore: Refactor package xyz` format. The Jira ticket code was included if available.
- [ ] PR has a focused scope and doesn't mix features with refactoring
- [ ] Tests are included (if applicable)
- [ ] `CHANGELOG.md` is updated (if applicable)
- [ ] CONFIG/SECRETS changes are updated in helmcharts and deployments (if applicable)
- [ ] Preview deployment works as expected
- [ ] Ready for production

### CODE REVIEW

File: internal/serve/httphandler/embedded_wallet_profile_handler_test.go [hunk 1]
```diff
@@ -93,6 +93,98 @@ func Test_EmbeddedWalletProfileHandler_GetProfile(t *testing.T) {
...
+				var body EmbeddedWalletReceiverContactResponse
+				require.NoError(t, json.Unmarshal(resp.Body.Bytes(), &body))
+				require.NotNil(t, body.ReceiverContact)
+				assert.Equal(t, ttc.expected.Type, body.ReceiverContact.Type)
+				assert.Equal(t, ttc.expected.Value, body.ReceiverContact.Value)
+			})
+		}
+	})
+
+	t.Run("unauthorized when receiver contact missing", func(t *testing.T) {
```
Comment (philipliu) [inline internal/serve/httphandler/embedded_wallet_profile_handler_test.go]: We should add a test for an empty contact address in the context.

File: internal/serve/serve_test.go [hunk 2]
```diff
@@ -706,6 +706,7 @@ func Test_handleHTTP_authenticatedEndpoints(t *testing.T) {
 		// Embedded wallets
 		{http.MethodGet, "/embedded-wallets/profile"},
 		{http.MethodGet, "/embedded-wallets/profile-assets"},
+		{http.MethodGet, "/embedded-wallets/profile-receiver"},
```
[the remark this answers is not in the document]
Comment (JiahuiWho) [reply internal/serve/serve_test.go]: Honest I’d like to merge the three profile endpoints into a single to keeps the wallet metadata in sync and saves clients multiple round-trips.

File: internal/serve/httphandler/embedded_wallet_profile_handler.go [hunk 3]
```diff
@@ -89,8 +124,36 @@ func (h EmbeddedWalletProfileHandler) GetAssets(rw http.ResponseWriter, req *htt
...
+	if err == nil {
+		return false
+	}
+
+	switch {
+	case errors.Is(err, services.ErrMissingContractAddress), errors.Is(err, data.ErrRecordNotFound):
+		httperror.Unauthorized("", err, nil).Render(rw)
+	default:
+		httperror.InternalError(ctx, internalErrMsg, err, nil).Render(rw)
+	}
```
Comment (philipliu) [inline internal/serve/httphandler/embedded_wallet_profile_handler.go]: Can we inline the error handling here? I’m not sure every call this wrapper covers returns `ErrRecordNotFound`, and even if it does, I don’t think we should always map that to `Unauthorized`. In some cases, it probably should be like a `NotFound` (or a different error) depending on context.

**Review by philipliu — APPROVED** [review]
